meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -0,0 +1,320 @@
+                                                });
+                                                $.getJSON(api_root + locationAPI, function(data) {
+                                                    $.each(data, function(index, value) {
+                                                        getClinicCaseAndType(value);
+                                                    });
+                                                    //Append the table and the columns ...
+                                                    $('#div_tabletCompletness').append('<table  id="tbleTabletSubmission"></table>');
+                                                    $('#div_tabletCompletness table').bootstrapTable({
+                                                        columns: columnNameArray,
+                                                        data: tabletDataArray,
+                                                        pagination: true,
+                                                        pageSize: 10,
+                                                        search: true,
+                                                        classes: "table table-no-bordered table-hover table-responsive"
+                                                    });
+                                                    addPaginationListener('#clinicsTable table');
+                                                    colorClinicsRow();
+                                                });
+
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
+}
+
+
+function buildTableStructure() {
+    //Add the first columns
+    buildTableColumn("DevicesIDs", "Devices IDs", "");
+    buildTableColumn("clinicType", "clinic type", "");
+    buildTableColumn("caseType", "case type", "");
+
+    for (i = week - 2; i <= week; i++) {
+        //Draw the clolumns depend on the week number
+        buildTableColumn("casesInWeek" + i, "cases in Week " + i, "");
+    }
+
+    for (i = week - 2; i <= week; i++) {
+        //Draw the clolumns depend on the week number
+        buildTableColumn("noOfRegistersInWeek" + i, "no. of registers in week " + i, "");
+    }
+
+    buildTableColumn("casesInMonthC", "cases in month (current)", "");
+    buildTableColumn("casesInMonthL", "cases in month (before the current)", "");
+
+    for (i = current_Year - 2; i <= current_Year; i++) {
+        //Draw the clolumns depend on the year number ...
+        buildTableColumn("casesInYear" + i, "cases in year " + i, "");
+    }
+}
+
+
+//This function is responsible for building the table header
+function buildTableColumn(columnId, columnName, columnClass) {
```

Comment:
The comment says this method is building table header while functions name is saying its building the whole column.
Please change the function name to say what the function does and afterwards you can remove the comment.

Reply:
edited